single_kernel_mongo/utils/mongo_connection.py

```diff
@@ -0,0 +1,679 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""Definition of MongoDB Connections."""
```

Comment:
re: my earlier comment about mongo manager, maybe the functions that are used by other managers should go here?

Reply:
This is purely for mongo interactions, nothing related to managers.
The flow is the following:
```
<event> -> <manager> -> <actual update on mongo / fs>
            |-> <write in DB>
```
I do agree that we might want to split the mongo manager in two parts, but I think this can be for later.
We don't have much time left for this project and I'd love to be able to finish in time :)

Reply:
Okay no worries, thank you!